File: CS/WebSite/Models/Customer.cs
using System;
using System.Collections.Generic;

namespace E2844.Models {
    public class Customer {
        public Customer() {
            ID = -1;
            Name = string.Empty;
            Country = -1;
            City = -1;
        }

        public int ID { get; set; }
        public string Name { get; set; }
        public int Country { get; set; }
        public int City { get; set; }
    }

    
    public class Country {
        public int ID { get; set; }
        public string Name { get; set; }

        public static IEnumerable<Country> GetCountries(){
            List<Country> list = new List<Country>();
            for(int i = 0; i < 100; i++)
                list.Add(new Country{ID=i, Name="Country" + i.ToString()});
            return list;
        }
    }
    public class City {
        public int ID { get; set; }
        public string Name { get; set; }
        public int CountryID { get; set; }

        public static IEnumerable<City> GetCities(int country) {
            List<City> list = new List<City>();
            for(int i = 0; i < 10000; i++)
                if(country >= 0 && i % 100 == country)
                    list.Add(new City { ID = i, Name = "City" + i.ToString(), CountryID = i % 100 });
            return list;
        }
    }
}

File: VB/DevExpressMvc3CascadingCombo/Models/Customer.vb
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Web
Imports System.Web.Mvc

Namespace CS.Models
	Public Class Customer
		Public Sub New()
			ID = -1
			Name = String.Empty
			Country = -1
			City = -1
		End Sub

		Public Property ID() As Integer
		Public Property Name() As String
		Public Property Country() As Integer
		Public Property City() As Integer
	End Class


	Public Class Country
		Public Property ID() As Integer
		Public Property Name() As String

		Public Shared Function GetCountries() As IEnumerable(Of Country)
			Dim list As New List(Of Country)()
			For i As Integer = 0 To 99
				list.Add(New Country With {.ID = i, .Name = "Country" & i.ToString()})
			Next i
			Return list
		End Function
	End Class
	Public Class City
		Public Property ID() As Integer
		Public Property Name() As String
		Public Property CountryID() As Integer

		Public Shared Function GetCities(ByVal c As Integer) As IEnumerable(Of City)
			Dim list As New List(Of City)()
			For i As Integer = 0 To 9999
				If c >= 0 AndAlso i Mod 100 = c Then
					list.Add(New City With {.ID = i, .Name = "City" & i.ToString(), .CountryID = i Mod 100})
				End If
			Next i
			Return list
		End Function
	End Class
End Namespace
